miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,769 @@
+        tz/tzs= time-zone
+        """
+        return self.data["timing"]
+
+    @property
+    def carpet_mode(self) -> bool:
+        """Auto boost on carpet."""
+        return self.data["auto_boost"]
+
+    def _parse_forbid_mode(self, val) -> DNDStatus:
+        # Example data: {"time":[75600,21600,1],"tz":2,"tzs":7200}
+        def _seconds_to_components(val):
+            hour = math.floor(val / 3600)
+            minut = math.floor((val - hour * 3600) / 60)
+            return (hour, minut)
+
+        as_dict = json.loads(val)
+        enabled = bool(as_dict["time"][2])
+        start = _seconds_to_components(as_dict["time"][0])
+        end = _seconds_to_components(as_dict["time"][1])
+        return DNDStatus(
+            dict(
+                enabled=enabled,
+                start_hour=start[0],
+                start_minute=start[1],
+                end_hour=end[0],
+                end_minute=end[1],
+            )
+        )
+
+    @property
+    def dnd_status(self) -> DNDStatus:
+        """Returns do-not-disturb status."""
+        return self._parse_forbid_mode(self.data["forbid_mode"])
+
+    @property
+    def water_level(self) -> WaterLevel:
+        """Get current water level."""
+        try:
+            return WaterLevel(self.data["water_level"])
+        except ValueError:
+            _LOGGER.error("Unknown WaterLevel (%s)", self.data["water_level"])
+            return WaterLevel.Unknown
+
+    @property
+    def double_clean(self) -> bool:
+        """Is double clean enabled."""
+        return self.data["double_clean"]
+
+    @property
+    def edge_sweep(self) -> bool:
+        """Is edge clean enabled."""
+        return self.data["edge_sweep"]
+
+    @property
+    def vacuum_led(self) -> bool:
+        """Return if led/display on vaccum is on."""
+        return self.data["led_switch"]
+
+    @property
+    def is_lidar_collision_sensor(self) -> bool:
+        """When ON, the robot will use lidar as the main detection sensor to help reduce
+        collisions."""
+        return self.data["lidar_collision"]
+
+    @property
+    def station_key(self) -> bool:
+        """When ON: long press the display will turn on dust collection."""
+        return self.data["station_key"]
+
+    @property
+    def station_led(self) -> bool:
+        """Return if station display is on."""
+        return self.data["station_led"]
+
+    @property
+    def current_audio(self) -> str:
+        """Current voice setting.
+
+        E.g. 'girl_en'
+        """
+        return self.data["current_audio"]
+
+    @property
+    def clean_time(self) -> timedelta:
+        """Time used for cleaning (if finished, shows how long it took)."""
+        return timedelta(seconds=self.data["clean_time_sec"])
+
+    @property
+    def clean_area(self) -> int:
+        """Cleaned area in m2."""
+        return self.data["clean_area"]
+
+    @property
+    def progress(self) -> str:
+        """Always return None?"""
```

Comment:
Is this `None` or `"None"` (as a string)? I'd suggest either removing this if it is not working as expected, as that would otherwise confuse anyone trying to use it (causing issue reports on github for a broken feature :-).

Reply:
I have now tried this again. The returned value is the same as if the feature is in not implemented. I will remove this function, and just leave a comment in the _MAPPING dict